### PR TITLE
NIFI-10192 Cache schema request result for reuse

### PR DESCRIPTION
**Pull request by kent-nguyen**
# Summary

[NIFI-10192](https://issues.apache.org/jira/browse/NIFI-10192)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI-10192) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 8
  - [x] JDK 11
  - [x] JDK 17

### Licensing

- [x] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [x] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [x] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Comment by mattyb149** [comment]
Looks like you have some Checkstyle errors, please run your maven command with the `-Pcontrib-check` before pushing commits, thanks!

**Comment by chnorton** [comment]
Looks like all of the automated checks have passed - was there anything else that needs to be done before this can be accepted @mattyb149 ?

**Comment by ottobackwards** [comment]
@markap14 did the work  [5955](https://github.com/apache/nifi/pull/5955), he would be the best to understand the issue and the side effects, ie.  if this is the right fix or there needs to be an additional refactor which includes looking at the original fix again.

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/LookupRecord.java [hunk 1]
```diff
@@ -536,6 +536,8 @@ public int getLookupCount() {
     private class RecordPathReplacementStrategy implements ReplacementStrategy {
         private int lookupCount = 0;
 
+        private ArrayList SCHEMA_LOOKUP_RESULT_CACHE = new ArrayList();
```
Comment (mattyb149) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/LookupRecord.java:539]: Although this isn't a CheckStyle error, we usually reserve variable names in all caps to constants. Perhaps change to `schemaLookupResultCache`?

**Review by mattyb149 — CHANGES_REQUESTED** [review]
Previously, we pushed down the responsibility of caching lookup values to the lookup service itself (see `AbstractDatabaseLookupService` ) rather than the processor(s) that use them. I'm not opposed to having a cache in LookupRecord as well, but it should be configurable as is the one AbstractDatabaseLookupService. This way the user can decide whether to use caching, how many values to cache, and we can take care of clearing the cache before the processor is (re)started. In this PR the cache is never cleared and if the lookup values from the LookupService change, they will never get picked up by the processor.

**Comment by mattyb149** [comment]
Can this be closed if #6364 replaces it?

**Comment by Aerilym** [comment]
> Can this be closed if #6364 replaces it?

Yes, this one can be closed, I have taken over the PR from @kent-nguyen and implemented a modified solution based on the given feedback.

**Comment by exceptionfactory** [comment]
Closing based on comments, thanks @Aerilym!

**Pull request closed without merging**